tests/test_logging.py

```diff
@@ -0,0 +1,111 @@
+from datetime import datetime
+import os
+
+
+def test_log():
+    """Test basic log operation outside model operation"""
+    from tm2py.config import LoggingConfig
+    import tempfile
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        log_config = {
+            "display_level": "STATUS",
+            "run_file_path": "log_run.txt",
+            "run_file_level": "STATUS",
+            "log_file_path": "log.txt",
+            "log_file_level": "DEBUG",
+            "log_on_error_file_path": "error.txt",
+            "notify_slack": False,
+            "use_emme_logbook": False,
+            "iter_component_level": None,
+        }
+
+        class Config:
+            logging = LoggingConfig(**log_config)
+
+        class Controller:
+            config = Config()
+            run_dir = temp_dir
+            iter_component = None
+
+        class TestException(Exception):
+            pass
+
+        controller = Controller()
+
+        from tm2py.logger import Logger
+
+        try:
+            with Logger(controller) as logger:
+                logger.log("a message")
+                logger.log("A status", level="STATUS")
+                logger.clear_msg_cache()
+                with logger.log_start_end("Running a set of steps"):
+                    logger.log_time("Indented message with timestamp")
+                    logger.log_time(
+                        "Indented displayed message with timestamp", level="STATUS"
+                    )
+                    logger.log(
+                        "A detail message, no time, but indented",
+                        level="DEBUG",
+                        indent=True,
+                    )
+                    logger.log("A debug message", level="DEBUG")
+                    logger.log("A trace message", level="TRACE")
+                    if logger.debug_enabled:
+                        # only generate this report if logging DEBUG
+                        logger.log(
+                            "A debug report that takes time to produce", level="DEBUG"
+                        )
+                raise TestException("an error")
+        except TestException:
+            pass
+
+        with open(os.path.join(temp_dir, log_config["run_file_path"]), "r") as f:
```

Comment:
I"m not exactly sure what this is testing? Might be helpful to add comments.

Reply:
Added some comments, checking the content of the log files, hopefully clear for the future.